src/libraries/System.ObjectModel/src/System/Collections/CollectionHelpers.cs

```diff
@@ -0,0 +1,73 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+
+namespace System.Collections
+{
+    internal static class CollectionHelpers
+    {
+        internal static void ValidateArguments(int sourceCount, Array array, int index)
```

Comment:
Nit: naming feels kind of ambiguous to me. How about `ValidateTargetArray` or `ValidateCopyToArguments`?

Reply:
Also, this looks like a good candidate for reuse in System.Collections.Generic implementations.

Reply:
> Also, this looks like a good candidate for reuse in System.Collections.Generic implementations.

Yes, there are a bunch of places across a variety of projects that could consolidate to use the same function.  I'd prefer we do that separately.

Reply:
> naming feels kind of ambiguous to me. How about ValidateTargetArray or ValidateCopyToArguments?

Ok, I'll change it to ValidateCopyToArguments.